boards/arm/sparkfun_pro_micro_rp2040/board.cmake

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# This configuration allows selecting what debug adapter debugging to use
+# for the SparkFun Pro Micro RP2040 by a command-line argument.
+# It is mainly intended to support both the 'picoprobe' and 'raspberrypi-swd'
+# adapter described in "Getting started with Raspberry Pi Pico".
+# And any other SWD debug adapter might also be usable with this configuration.
+
+# Note that the Pro Micro RP2040 SWD pins are test points on the underside
+# of the board.
+
+# Set RPI_PICO_DEBUG_ADAPTER to select debug adapter by command-line arguments.
+# e.g.) west build -b rpi_pico -- -DRPI_PICO_DEBUG_ADAPTER=raspberrypi-swd
+# The value is treated as a part of an interface file name that
+# the debugger's configuration file.
+# The value must be the 'stem' part of the name of one of the files
+# in the openocd interface configuration file.
+# The setting is store to CMakeCache.txt.
+if ("${RPI_PICO_DEBUG_ADAPTER}" STREQUAL "")
+	set(RPI_PICO_DEBUG_ADAPTER "cmsis-dap")
+endif()
+
+board_runner_args(openocd --cmd-pre-init "source [find interface/${RPI_PICO_DEBUG_ADAPTER}.cfg]")
+board_runner_args(openocd --cmd-pre-init "transport select swd")
+board_runner_args(openocd --cmd-pre-init "source [find target/rp2040.cfg]")
+
+# The adapter speed is expected to be set by interface configuration.
+# But if not so, set 2000 to adapter speed.
+board_runner_args(openocd --cmd-pre-init "set_adapter_speed_if_not_set 2000")
```

Comment:
Also you need to copy
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/rpi_pico/support/openocd.cfg.

Reply:
Done